hazelcast/src/main/java/com/hazelcast/spi/impl/SlowOperationLog.java

```diff
@@ -208,8 +211,10 @@ public JsonObject toJson() {
         public void fromJson(JsonObject json) {
             id = getInt(json, "id");
             startedAt = getLong(json, "startedAt");
-            durationNanos = getLong(json, "duration");
             lastAccessNanos = System.nanoTime();
+
+            // write durationMs last to enforce a write barrier
+            durationMs = getInt(json, "durationMs");
```

Comment:
The write to `durationMs` should piggy back the changes to the other variables, too. And since the order should be important I think it's best to document this, so nobody changes it accidentally.